### PR TITLE
[EDITORS and IDEs] adding description of Atom editor

### PR DESCRIPTION
**Pull request by mounalbaccouch**
Updating the following file to add link to Atom description: dev/source/docs/code-editing-tools-and-ides.rst 
Adding the following file for Atom description: dev/source/docs/editing-the-code-with-atom.rst

### CODE REVIEW

File: dev/source/docs/editing-the-code-with-atom.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Atom is a free and open-source editor available for Linux, MacOS and Windows developed by GitHub.
+It is a perfect complement to GitHub's primary mission of building better software by working together.
+
+It bundles all tools needed to build ArduPilot : C/C++ support, Python support, Git/GitHub support, :ref:`debugging support <debugging-with-gdb-on-linux>`.
+Using Teletype package, Atom lets developers share their workspace with team members and collaborate on code in real time.
+
+The complete manual is available `here <https://flight-manual.atom.io/>`__.
+To install Atom, follow the instructions available `here <https://flight-manual.atom.io/getting-started/sections/installing-atom/>`__.
+To configure the GitHub package, follow the instructions available `here <https://flight-manual.atom.io/using-atom/sections/github-package/>`__.
+Toc onfigure Teletype package, follow the instructions available `here <https://teletype.atom.io/>`__.
```
Comment (khancyr) [inline dev/source/docs/editing-the-code-with-atom.rst:16]: To configure

File: dev/source/docs/editing-the-code-with-atom.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.. _editing-the-code-with-atom:
+
+============================
+Editing the code with Atom
+============================
```
Comment (khancyr) [inline dev/source/docs/editing-the-code-with-atom.rst:5]: === should be the length of the title

**Review by khancyr — CHANGES_REQUESTED** [review]
Could you add the full instruction to be able to use Atom ? which package we need, how to configure to build, etc ?
For VSCode we don't need that as VSCode is Atom + C/C++ + python support

File: dev/source/docs/editing-the-code-with-atom.rst (hunk 1 above)
Comment (mounalbaccouch) [reply dev/source/docs/editing-the-code-with-atom.rst]: I prefer Atom because it is more simple and supports GitHub. 
I updated the page as recommended.

**Comment by mounalbaccouch** [comment]
Can you please link this issue with this PR please ?
https://github.com/ArduPilot/ardupilot_wiki/issues/3206

**Comment by Hwurzburg** [comment]
> Can you please link this issue with this PR please ?
> #3206

 you did it as soon as you mentioned it

File: dev/source/docs/editing-the-code-with-atom.rst [hunk 3]
```diff
@@ -1,16 +1,80 @@
...
+
+You can visit the official GitHub manual for Atom `here <https://flight-manual.atom.io/using-atom/sections/github-package/>`__ to to start managing your GitHub repositories from Atom.
+
+Install Teletype package for real time collaboration
+====================================================
+
+Teletype allows sharing or joining a portal for real time collaboration. To install it, go to the menu bar *Packages->Settings View->Open*.
+Type in "Teletype" in the box then install. Finally, you can see Teletype in the menu *Packages* then you can *"Join Portal"* or *"Share Portal"*.
+
+.. image:: ../images/atom_teletype.png
```
Comment (Hwurzburg) [inline dev/source/docs/editing-the-code-with-atom.rst:75]: incorrect links:    should be .. images:: ../../../images/atom_teletype.png    same for github token image ... or you can place them in the DEV site's images folder

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
atom
I suggest adding a screenshot like this at the top of the page also

File: dev/source/docs/editing-the-code-with-atom.rst (hunk 3 above)
Comment (mounalbaccouch) [reply dev/source/docs/editing-the-code-with-atom.rst:78]: Thanks for your notes.
I added your screenshot at the top of the page.

**Comment by Hwurzburg** [comment]
I fixed a bunch of "line block" marks that result in build errors....squashed....will merge....thanks!

**Pull request merged**